### DISCUSSION TITLE
Pulling rulesets from a private github repository

### DISCUSSION THREAD

**Discussion by kevin-kortum-trustedshops** [terraform-linters/tflint#2142] [Q&A]
Hi everyone,
i am currently trying to pull a ruleset from a private github repository of which i am not the owner but i do have access to it with my user. However, even when i set up the GITHUB_TOKEN env var, `tflint --init` does not seem to be able to pull the ruleset.

`Failed to install a plugin; Failed to fetch GitHub releases: GET https://api.github.com/repos/my-comany/tflint-ruleset-company-rules/releases/tags/v1.0.0: 404 Not Found []`

Has anyone got this working already or could this be a bug?

**Comment by kevin-kortum-trustedshops** [comment]
Found a discussion here https://github.com/terraform-linters/tflint/discussions/2004

**Reply by wata727** [reply]
Note that the discussion you are referring to is about how to use both private repo in github.com and GitHub Enterprise at the same time, not a limitation on working with private repo rulesets. Moreover, this issue has already been fixed https://github.com/terraform-linters/tflint/discussions/2004#discussioncomment-11008051

From what I understand, with a properly set `GITHUB_TOKEN`, it should be possible to install even from private repositories.

**Comment by bendrucker** [comment]
You should always authenticate your requests to the GitHub API, to avoid the low per-IP rate limit imposed on anonymous requests:

https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/plugins.md#avoiding-rate-limiting

#2005 covers providing _different_ credentials to _different_ GitHub servers (i.e. GitHub Enterprise Server). It doesn't change anything if you only use `github.com`.

There's a bug in your workflow, either the repo/tag is incorrect or you aren't passing the credentials as documented:

https://github.com/terraform-linters/setup-tflint?tab=readme-ov-file#usage
